src/components/income/F1099Info.tsx

```diff
@@ -53,29 +70,72 @@ function List1099s (): ReactElement {
 }
 
 interface F1099UserInput {
-  payer: string
-  income: string
   formType: Income1099Type
+  payer: string
+  // Int fields
+  interest: string
+  // B Fields
+  shortTermProceeds: string
+  shortTermCostBasis: string
+  longTermProceeds: string
+  longTermCostBasis: string
+  // Div fields
+  dividends: string
+  qualifiedDividends: string
   personRole: PersonRole.PRIMARY | PersonRole.SPOUSE
 }
 
-const toF1099 = (formData: F1099UserInput): Income1099 => ({
-  ...formData,
-  // Note we are not error checking here because
-  // we are already in the input validated happy path
-  // of handleSubmit.
-  income: parseInt(formData.income)
-})
+const toF1099 = (input: F1099UserInput): Supported1099 => {
+  switch (input.formType) {
+    case Income1099Type.INT: {
+      return {
+        payer: input.payer,
+        personRole: input.personRole,
+        type: input.formType,
+        form: {
+          income: parseInt(input.interest)
+        }
+      }
+    }
+    case Income1099Type.B: {
+      return {
+        payer: input.payer,
+        personRole: input.personRole,
+        type: input.formType,
+        form: {
+          shortTermCostBasis: parseInt(input.shortTermCostBasis),
```

Comment:
I think 2 decimal places should be supported for these fields because the final rounded integer profit of the transaction can vary based on the decimal values that are subtracted. Ex 2-1 is rounded to 1,  2.00 - 1.50 is rounded down to 0

Could use something like
`(Math.round(num * 100) / 100).toFixed(2);`

Reply:
Agree but the same problem exists for the W2 and remember we are using the same currency pattern everywhere:

```
currency: {
    regexp: /^[0-9]+$/,
    description: 'Input should be filled with numbers only'
  },
```

Which doesn't accept decimals. So I think this is a separate issue now.

Reply:
That makes sense. I think this issue would address the problem https://github.com/ustaxes/UsTaxes/issues/18